data/rest-data_test.go

```diff
@@ -119,3 +120,60 @@ func TestGetSubdirContentByPath(t *testing.T) {
 		assert.Contains(t, err.Error(), "no subdirectories found")
 	})
 }
+func TestIsCodeRepo(t *testing.T) {
+	tests := []struct {
+		name           string
+		responses      []mock.MockBackendOption
+		expectedResult bool
+		expectedError  bool
+	}{
+		{
+			name: "repository with code languages",
+			responses: []mock.MockBackendOption{
+				mock.WithRequestMatch(
+					mock.GetReposLanguagesByOwnerByRepo,
+					map[string]int{"Go": 1000, "JavaScript": 500},
+				),
+			},
+			expectedResult: true,
+			expectedError:  false,
+		},
+		{
+			name: "repository with no languages",
+			responses: []mock.MockBackendOption{
+				mock.WithRequestMatch(
+					mock.GetReposLanguagesByOwnerByRepo,
+					map[string]int{},
+				),
+			},
+			expectedResult: false,
+			expectedError:  false,
+		},
+		{
+			name:           "api error",
+			responses:      []mock.MockBackendOption{},
+			expectedResult: true,
+			expectedError:  true,
+		},
```

Comment:
The [docs on returning errors](https://github.com/migueleliasweb/go-github-mock?tab=readme-ov-file#mocking-errors-from-the-api) suggest this pattern

```suggestion
        {
			name: "api error",
			responses: []mock.MockBackendOption{
				mock.WithRequestMatchHandler(
					mock.GetReposLanguagesByOwnerByRepo,
					http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
						mock.WriteError(w, http.StatusInternalServerError, "github went belly up or something")
					}),
				),
			},

			expectedResult: true,
			expectedError:  true,
		},
```